SOURCES/xha-10.1.0-supports-drbd.XCP-ng.patch

```diff
@@ -0,0 +1,189 @@
+From 0c1da77c5e6a4bb8a7dd759b3ccd1beafe24b4cc Mon Sep 17 00:00:00 2001
+From: Ronan Abhamon <ronan.abhamon@vates.fr>
+Date: Thu, 6 Aug 2020 16:53:57 +0200
+Subject: [PATCH] feat(statefile): supports DRBD (close state file descriptor
+ after each write/read)
+
+---
+ daemon/statefile.c | 48 +++++++++++++++++++++++++++++++++++++++-------
+ lib/statefileio.c  | 29 +++++++++++++++++++++++++++-
+ 2 files changed, 69 insertions(+), 8 deletions(-)
+
+diff --git a/daemon/statefile.c b/daemon/statefile.c
+index af819a0..09ff1f2 100755
+--- a/daemon/statefile.c
++++ b/daemon/statefile.c
+@@ -73,6 +73,9 @@
+ //
+ //
+ 
++extern int
++is_drbd_device(
++      const char *path);
+ 
+ //
+ //
+@@ -122,8 +125,11 @@ static struct {
+ //  Internal data for this module
+ //
+ 
++#define SFVAR_FLAG_DRBD (1 << 0)
++
+ static struct {
+     int                 sfdesc;                 //  File descriptor for the State-File
++    int                 flags;
+     MTC_BOOLEAN         terminate;
+     MTC_U32             sequence;
+     pthread_t sf_thread;
+@@ -152,6 +158,17 @@ static struct {
+     } readlatency, writelatency;
+ } sfvar = { 0 };
+ 
++static inline int open_desc()
++{
++    return sfvar.flags & SFVAR_FLAG_DRBD ? sf_open(_sf_path) : sfvar.sfdesc;
++}
++
++static inline void close_desc(int sfdesc)
++{
++    if (sfvar.flags & SFVAR_FLAG_DRBD)
++        sf_close(sfdesc);
++}
++
+ //  lock
+ #define sf_lock()   ((void)pthread_spin_lock(&sfvar.lock))
+ #define sf_unlock() ((void)pthread_spin_unlock(&sfvar.lock))
+@@ -303,6 +320,7 @@ sf_initialize0()
+     //  Initialize sfvar
+ 
+     sfvar.sfdesc = -1,
++    sfvar.flags = 0,
+     sfvar.watchdog = INVALID_WATCHDOG_HANDLE_VALUE,
+     sfvar.terminate = FALSE,
+ 
+@@ -407,15 +425,23 @@ sf_initialize0()
+ 
+     //  Open the State-File
+ 
+-    if ((sfvar.sfdesc = sf_open(_sf_path)) < 0)
++    const int sfdesc = sf_open(_sf_path);
++    if (sfdesc < 0)
+     {
+-        sfvar.sfdesc = -1;
+         status = MTC_ERROR_SF_OPEN;
+         log_internal(MTC_LOG_ERR,
+-                    "SF: cannot open the State-File %s.\n", _sf_path);
++                    "SF: cannot open the State-File %s. (sys %d)\n", _sf_path, errno);
+         goto error;
+     }
+ 
++    if (is_drbd_device(_sf_path))
++    {
++        sfvar.flags |= SFVAR_FLAG_DRBD;
++        sf_close(sfdesc);
++    }
++    else
++        sfvar.sfdesc = sfdesc;
```

Comment:
Does xha's coding style allow to use `{}` for the if block but not for the else block?